src/vm/analysis/read_only_checker/mod.rs

```diff
@@ -229,12 +229,16 @@ impl <'a, 'b> ReadOnlyChecker <'a, 'b> {
             },
             ContractCall => {
                 check_arguments_at_least(2, args)?;
-                let contract_name = args[0].match_atom()
-                    .ok_or(CheckErrors::ContractCallExpectName)?;
+                let contract_identifier = match args[0].expr {
+                    SymbolicExpressionType::AtomValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))) => contract_identifier,
+                    SymbolicExpressionType::LiteralValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))) => contract_identifier,
```

Comment:
We should match _only_ a Literal here, and error otherwise. Once sugar expander is changed to only produce Literals rather than AtomValues, I don't think that the AtomValue branch could even be visited, but we should enforce that with an error return anyways.

Reply:
If we are also transforming the fully qualified to LiteralValues, then yes indeed, this branch should throw 👍

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/c871057b86ad7ab2c64a84ba2986f42b469860ec